apps/external_ledger_db/lib/external_ledger_db/wallet.ex

```diff
@@ -0,0 +1,174 @@
+# Copyright 2017-2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule ExternalLedgerDB.Wallet do
+  @moduledoc """
+  Ecto Schema representing external ledger wallets. A wallet is made up of
+  a unique address and the ID associated with it.
+  """
+  use Ecto.Schema
+  import Ecto.{Changeset, Query}
+  alias ExternalLedgerDB.Repo
+
+  @behaviour Utils.Ledgers.WalletSchema
+
+  @primary_key {:uuid, Ecto.UUID, autogenerate: true}
+  @timestamps_opts [type: :naive_datetime_usec]
+
+  @hot "hot"
+  @cold "cold"
+
+  def hot, do: @hot
+  def cold, do: @cold
+
+  @ethereum "ethereum"
+  @omg_network "omg_network"
+
+  def ethereum, do: @ethereum
+  def omg_network, do: @omg_network
```

Comment:
This shouldn't be hardcoded. The adapters should have a way to register themselves.